### PR TITLE
docs: A tutorial on how to overload converters in Torch-TensorRT

### PR DESCRIPTION
**Pull request by narendasan**
# Description

Small tutorial on how to overload converters in Torch-TensorRT

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: examples/dynamo/converter_overloading.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+import logging
+import sys
+
+import torch
+import torch_tensorrt
+
+# %% GeLU Operator in PyTorch
+#
+# GeLU has 2 modes in PyTorch, one using the ``erf`` function and the other using the ``tanh`` approximation.
+# TensorRT natively supports both implementations as am activation layer, but suppose we want to use a custom implementation of GeLU in TensorRT only for ``tanh`` mode.
```
Comment (zewenli98) [inline examples/dynamo/converter_overloading.py:25]: am -> an

**Review by peri044 — APPROVED** [review]
LGTM.

File: examples/dynamo/converter_overloading.py (hunk 1 above)
Comment (peri044) [inline examples/dynamo/converter_overloading.py:25]: nit: as an activation layer

File: examples/dynamo/converter_overloading.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+# This is useful for cases where you want to use a custom converter only in specific cases, like in our case where we only want to use our converter when ``approximate == "tanh"``.
+#
+# Distinct to the validator is the ``supports_dynamic_shapes`` argument, which is a boolean indicating if the converter can be used in cases where the input shapes are dynamic.
+# If this is set to ``False``, in cases where the inputs provided by the user are dynamic, this converter will be disabled. If there are no alternatives that support dynamic shape, the operation will be run in PyTorch.
+#
+# Finally there is the ``priority`` argument, which is an enum from the ``torch_tensorrt.dynamo.conversion.ConverterPriority`` class that defines the priority of the converter. The two options are ``HIGH`` and ``STANDARD``.
+# Converters registered with ``STANDARD`` will be appended to the converter list for a given operation, while converters registered with ``HIGH`` will be prepended to the list.
+# Canidate converters are evalated for their suitablity in this priority order and the first converter that passes the validator is used.
```
Comment (zewenli98) [inline examples/dynamo/converter_overloading.py:101]: Canidate -> Candidate

File: examples/dynamo/converter_overloading.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+
+# %%
+# Converter Implementation
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# The converter function itself takes the following arguments: the current conversion context, the target operator, the arguments to the target operator, the keyword arguments to the target operator, and the name of the target operator.
+# Arguments can either any of python primitives, ``torch.Tensor``, ``np.Arrays`` or ``ITensor`` objects.
+# The converter function should return the outputs of the target operator in terms of TensorRT ``ITensor`` primarily.
+#
+# Since Torch-TensorRT covers the core ATen opset, it has already abstracted many of the common low-level operations into helper functions that can be used to build up the TensorRT network.
```
Comment (zewenli98) [inline examples/dynamo/converter_overloading.py:111]: Do we need to mention where users can find the ATen schema so that they can get to know the args' positions and types?

**Review by zewenli98 — COMMENTED** [review]
Overall LGTM, just some minor comments.
